egs/rm/s5/local/prepare_wsj_rm_lang.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Copyright 2017 Pegah Ghahremani
+
+# This script prepares a dictionary for wsj to rm transfer learning experiment
+# which uses wsj phone set, lexicon and dict and
+# the lexicon for rm words.txt are copied from wsj lexicon for common words in wsj
+# and rm. words in rm that are not in the wsj lexicon are added
+# as oov in lexicon.txt.
+# The oov word "<SPOKEN_NOISE>" in wsj is also added to words.txt and G.fst is recompiled using
+# updated word list.
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. utils/parse_options.sh
+
+if [ $# != 3 ]; then
+  echo "Usage: local/prepare_wsj_rm_lang.sh <src-dict> <src-lang> <output-dir>"
+  echo "e.g:"
+  echo "$0 ../../wsj/s5/data/local/dict ../../wsj/s5/data/lang_nosp data/wsj_rm_dir"
+fi
+
+src_dict=$1
+src_lang=$2
+output_dir=$3
+
+required_dict_files="$src_dict/lexicon.txt $src_dict/nonsilence_phones.txt $src_dict/silence_phones.txt $src_dict/optional_silence.txt $src_lang/oov.txt $src_lang/phones.txt"
+for f in $required_dict_files; do
+  if [ ! -f $f ]; then
+    echo "file $f that is required for preparing lang does not exists." && exit 1;
+  fi
+done
+
+rm -r $output_dir 2>/dev/null
+mkdir -p $output_dir
+mkdir -p $output_dir/local
+# copy *phones.txt from source to target.
+cp -r $src_dict $output_dir/local/dict
+rm $output_dir/local/dict/lexicon*.txt
+
+oov_word=`cat $src_lang/oov.txt`
+# common word list in rm lexicon with lexicon in wsj
+comm -12 <(awk '{print $1}' data/local/dict/lexicon.txt | sed "s/\+/\'/g" | sort) \
```

Comment:
I don't think the previous comments about this file were addressed. You can do this more easily using perl instead of multiple sort statements.